File: src/main/java/superstitio/cards/lupa/BaseCard/DrySemen.java
package superstitio.cards.lupa.BaseCard;

import basemod.cardmods.RetainMod;
import basemod.helpers.CardModifierManager;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import superstitio.DataManager;
import superstitio.cardModifier.modifiers.block.DrySemenBlock;
import superstitio.cards.lupa.LupaCard;

public class DrySemen extends LupaCard {
    public static final String ID = DataManager.MakeTextID(DrySemen.class);

    public static final CardType CARD_TYPE = CardType.SKILL;

    public static final CardRarity CARD_RARITY = CardRarity.UNCOMMON;

    public static final CardTarget CARD_TARGET = CardTarget.SELF;

    private static final int COST = 1;
    private static final int BLOCK = 12;
    private static final int UPGRADE_BLOCK = 5;
    private static final int MAGIC = 3;
    private static final int UPGRADE_MAGIC = 0;

    public DrySemen() {
        super(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET);
        this.setupBlock(BLOCK, UPGRADE_BLOCK, new DrySemenBlock());
        this.setupMagicNumber(MAGIC, UPGRADE_MAGIC);
        CardModifierManager.addModifier(this, new RetainMod());
    }

    @Override
    public void use(AbstractPlayer player, AbstractMonster monster) {
        if (hasEnoughSemen(this.magicNumber)) {
            useSemen(this.magicNumber);
            addToBot_gainBlock();
        }
    }

    @Override
    public boolean canUse(AbstractPlayer p, AbstractMonster m) {
        return super.canUse(p, m) && hasEnoughSemen(this.magicNumber);
    }

    @Override
    public void upgradeAuto() {
    }
}

File: superstitioMain/java/superstitio/cards/lupa/BaseCard/DrySemen.kt
package superstitio.cards.lupa.BaseCard

import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.monsters.AbstractMonster
import superstitio.DataManager
import superstitio.cardModifier.modifiers.block.DrySemenBlock
import superstitio.cards.IsLupaCard
import superstitio.cards.lupa.LupaCard
import superstitioapi.cards.addRetainMod
@IsLupaCard
class DrySemen : LupaCard(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET, "base")
{
    init
    {
        this.setupBlock(BLOCK, UPGRADE_BLOCK, DrySemenBlock())
        this.setupMagicNumber(MAGIC, UPGRADE_MAGIC)
        this.addRetainMod()
    }

    override fun use(player: AbstractPlayer?, monster: AbstractMonster?)
    {
        if (hasEnoughSemen(this.magicNumber))
        {
            addToBot_useSemenAndAutoRemove(this.magicNumber)
            addToBot_gainBlock()
        }
    }

    override fun canUse(p: AbstractPlayer?, m: AbstractMonster?): Boolean
    {
        return super.canUse(p, m) && hasEnoughSemen(this.magicNumber)
    }

    override fun upgradeAuto()
    {
    }

    companion object
    {
        val ID: String = DataManager.MakeTextID(DrySemen::class.java)

        val CARD_TYPE: CardType = CardType.SKILL

        val CARD_RARITY: CardRarity = CardRarity.BASIC

        val CARD_TARGET: CardTarget = CardTarget.SELF

        private const val COST = 1
        private const val BLOCK = 13
        private const val UPGRADE_BLOCK = 5
        private const val MAGIC = 3
        private const val UPGRADE_MAGIC = 0
    }
}

